### DISCUSSION TITLE
Randomly returns N results

### DISCUSSION THREAD

**Discussion by youfeed** [meilisearch/product#601] [Feedback & Feature Proposal]
Hi, can I query random returns?
This is very necessary, it would be better if there are relevant recommendations

**Comment by oluademola** [comment]
Hi @youfeed 

I am not exactly sure I understand your question. Could you provide an example of what you mean?

**Comment by kossa** [comment]
Looking for the same thing, I want to make a search, then the result need to be sort randomly, dont want to show always the same items

Something like:
```sql
SELECT *
FROM jobs
WHERE is_featured = 1
order by RAND()
```

**Comment by oluademola** [comment]
Hello @kossa,

Thank you for the much-needed context. Meilisearch's philosophy is to return relevant search results. Indeed, we mentioned in the [dedicated documentation](https://docs.meilisearch.com/learn/what_is_meilisearch/philosophy.html#highly-customizable) that `"It wouldn't be a search engine if there wasn't a notion of relevancy in the results returned."` I am afraid this isn't the best use case for the engine and isn't possible natively. However, a viable workaround could be to transform your dataset by including an additional field with a random number in all your documents such that you can sort with this field to achieve a randomized result. Maybe @gmourier might have some suggestions.

I hope you find this helpful.

Thanks.

**Reply by kossa** [reply]
Thank you for you answer, currently I'm getting 100 result, I shuffle them, then I select 5, I want to optimize this, instead of selecting 100, why not only 5 :)

**Comment by dohankk** [comment]
any progress here?

**Comment by conradfr** [comment]
@oluademola but returning n random results is more about the "sort" than the "relevancy" part isn't it? And Meilisearch has a sort parameter.
